### PR TITLE
Fix 0D tensors

### PR DESCRIPTION
**Pull request by Happypig375**

### CODE REVIEW

File: UnitTests/Constructors.cs [hunk 1]
```diff
@@ -70,6 +70,49 @@ public void CreateVector()
...
+            x = GenTensor<int, IntWrapper>.Deserialize(new byte[] { 0, 0, 0, 0, 4, 0, 0, 0, 9, 0, 0, 0 });
+            Check(9);
+            //x = GenTensor<int, IntWrapper>.PiecewiseAdd(x, x);
+            //Check(18);
+            //x = GenTensor<int, IntWrapper>.PiecewiseMultiply(x, x);
+            //Check(324);
+            //x = GenTensor<int, IntWrapper>.PiecewiseDivide(x, x);
+            //Check(1);
+            //x = GenTensor<int, IntWrapper>.PiecewiseSubtract(x, x);
+            //Check(0);
```
Comment (Happypig375) [inline UnitTests/Constructors.cs:114]: I don't really know how these may be fixed

File: UnitTests/Constructors.cs [hunk 2]
```diff
@@ -70,6 +70,49 @@ public void CreateVector()
...
+                Assert.AreEqual(value, x[System.Array.Empty<int>()]);
+                Assert.AreEqual(value, x.GetCell(System.Array.Empty<int>()));
+                Assert.AreEqual(value, x.GetValueNoCheck(System.Array.Empty<int>()));
+                CollectionAssert.AreEqual(System.Array.Empty<int>(), x.blocks);
+                CollectionAssert.AreEqual(new[] { value }, x.data);
+                Assert.AreEqual(0, x.LinOffset);
+                CollectionAssert.AreEqual(new byte[] { 0, 0, 0, 0, 4, 0, 0, 0, (byte)value, 0, 0, 0 }, x.Serialize());
+            }
+            Check(0);
+            x[System.Array.Empty<int>()] = 3;
```
Comment (WhiteBlackGoose) [inline UnitTests/Constructors.cs:95]: How did this one even work?

File: UnitTests/Constructors.cs (hunk 1 above)
Comment (WhiteBlackGoose) [reply UnitTests/Constructors.cs:114]: What error is it giving?

Comment (Happypig375) [reply UnitTests/Constructors.cs:114]: image

Comment (WhiteBlackGoose) [reply UnitTests/Constructors.cs:114]: Okay but if it's 0D can't you just manually add the values?

Comment (Happypig375) [reply UnitTests/Constructors.cs:114]: but GenericTensor is generic

Comment (Happypig375) [reply UnitTests/Constructors.cs:114]: hmm

**Review by WhiteBlackGoose — APPROVED** [review]

**Pull request merged**